### PR TITLE
feat(TimePicker): new component

### LINKED ISSUE #375: [Feat]: Add new TimePicker component
**Description**: Time Picker component allows users to easily select a specific time, typically using an intuitive visual interface like a clock or list of time values.

**Requirements:**

_Component Implementation:_

1. Implement the Time picker component in TypeScript.
2. Utilize SCSS with BEM methodology for styling.
3. Integrate design tokens from `@geneui/tokens` for consistent colors, font sizes, spacing, and button styles.
4. Support configuration options such as minimum, maximum times, range selection, ability to show times as disabled.
5. Ensure proper accessibility, including keyboard navigation and ARIA attributes.

_Unit Tests:_

1. Write unit tests using Jest and React Testing Library.
2. Ensure test coverage for all props, including rendering of different configurations (min, max, disabledTimes).
3. Test for edge cases.

_Storybook Stories:_

1. Create a Storybook story for the Time picker component.
2. Showcase different configurations (e.g., various min/max times, range selection, disabled times).
3. Create controls for the props so users can interact with the Time picker component in Storybook.
4. Document all props of the public interfaces.

Checklist:

- [ ] Component is implemented in TypeScript.
- [ ] Styling follows SCSS using BEM methodology.
- [ ] Design tokens from `@geneui/tokens` are used for colors, font sizes, spacing, and button styles.
- [ ] Unit tests written and cover different states and edge cases.
- [ ] Storybook stories created with examples of all props and configuration options.
- [ ] Figma design matches the component implementation.

### PR DESCRIPTION
**Pull request by alberthovhannisyan1**
### Chromatic Build Summary 
**Build URL:** https://www.chromatic.com/build?appId=66daee7859a843dd87bfb3d7&number=1576 
**Storybook URL:** https://66daee7859a843dd87bfb3d7-vkirgleiwy.chromatic.com/

### CODE REVIEW

File: src/components/molecules/TimePicker/components/PickerButton/PickerButton.scss [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+        &_small {
+            font-family: var(--guit-sem-font-caption-large-medium-font-family);
+            font-size: var(--guit-sem-font-caption-large-medium-font-size);
+            font-weight: var(--guit-sem-font-caption-large-medium-font-weight);
+            line-height: var(--guit-sem-font-caption-large-medium-line-height);
+        }
+    }
+
+    &_state {
```
Comment (GaroGabrielyan) [inline src/components/molecules/TimePicker/components/PickerButton/PickerButton.scss:27]: Please remove `state` modificator

File: src/components/molecules/TimePicker/components/PickerButton/PickerButton.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import React, { ComponentPropsWithoutRef, FC, memo } from "react";
+import classNames from "classnames";
+
+// Styles
+import "./PickerButton.scss";
+
+// Types
+import { TimePickerSizes } from "../../types";
+
+interface IPickerButtonProps extends ComponentPropsWithoutRef<"button"> {
```
Comment (GaroGabrielyan) [inline src/components/molecules/TimePicker/components/PickerButton/PickerButton.tsx:10]: Please remove `ComponentPropsWithoutRef<"button">` type

File: src/components/molecules/TimePicker/TimePicker.tsx [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+import classNames from "classnames";
+import { nanoid } from "nanoid/non-secure";
+
+import { Clock } from "@geneui/icons";
+
+// Components
+import Label from "@components/atoms/Label";
+
+// Styles
+import "./TimePicker.scss";
```
Comment (GaroGabrielyan) [inline src/components/molecules/TimePicker/TimePicker.tsx:11]: Please reorder the impors

File: src/components/molecules/TimePicker/TimePicker.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+};
+
+export const RangePickerStates: RangeStory = {
+    render: ({ label, ...props }) => {
+        const controls = definedProps(props);
+
+        return (
+            <div style={casesStyle}>
+                {rangePickerCases.map(({ key, RenderComponent, ...item }) => (
+                    <div key={`${key}-${JSON.stringify(controls.defaultValue)}`} style={rangeStoryStyle}>
```
Comment (GaroGabrielyan) [inline src/components/molecules/TimePicker/TimePicker.stories.tsx:227]: please remove JSON.stringify

**Review by GaroGabrielyan — CHANGES_REQUESTED** [review]
